ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -101,100 +105,141 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
-    allNetworksList.includes(chainId),
+  const [selectedChainIds, setSelectedChainIds] = useState<CaipChainId[]>(
+    defaultSelectedChainIds as CaipChainId[],
   );
 
-  const defaultSelectedChainIds =
-    supportedRequestedChainIds.length > 0
-      ? supportedRequestedChainIds
-      : defaultSelectedNetworkList;
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccountsWithCaipAccountId,
+  ) as MergedInternalAccountWithCaipAccountId[];
+
+  const requestedNamespaces = getAllNonWalletNamespacesFromCaip25CaveatValue(
+    requestedCaip25CaveatValue,
+  );
 
-  const [selectedChainIds, setSelectedChainIds] = useState(
-    defaultSelectedChainIds,
+  // all accounts that match the requested namespaces
+  const supportedAccountsForRequestedNamespaces = allAccounts.filter(
+    (account) => {
+      const {
+        chain: { namespace },
+      } = parseCaipAccountId(account.caipAccountId);
+      return requestedNamespaces.includes(namespace);
+    },
   );
 
-  const accounts = useSelector(getUpdatedAndSortedAccounts);
-  const evmAccounts = useMemo(() => {
-    return accounts.filter((account: InternalAccount) =>
-      isEvmAccountType(account.type),
-    );
-  }, [accounts]);
+  // all requested accounts that are found in the wallet
+  const supportedRequestedAccounts = requestedCaipAccountIds.reduce(
+    (acc, account) => {
+      const supportedRequestedAccount =
+        supportedAccountsForRequestedNamespaces.find(({ caipAccountId }) =>
+          isEqualCaseInsensitive(caipAccountId, account),
+        );
+      if (supportedRequestedAccount) {
+        acc.push(supportedRequestedAccount);
+      }
+      return acc;
+    },
+    [] as MergedInternalAccountWithCaipAccountId[],
+  );
 
-  const supportedRequestedAccounts = requestedAccounts.filter((account) =>
-    evmAccounts.find(({ address }) => isEqualCaseInsensitive(address, account)),
+  const defaultAccounts = getDefaultAccounts(
+    requestedNamespaces,
+    supportedRequestedAccounts,
+    supportedAccountsForRequestedNamespaces,
   );
 
-  const currentAccount = useSelector(getSelectedInternalAccount);
-  const currentAccountAddress = isEvmAccountType(currentAccount.type)
-    ? [currentAccount.address]
-    : []; // We do not support non-EVM accounts connections
-  const defaultAccountsAddresses =
-    supportedRequestedAccounts.length > 0
-      ? supportedRequestedAccounts
-      : currentAccountAddress;
-  const [selectedAccountAddresses, setSelectedAccountAddresses] = useState(
-    defaultAccountsAddresses,
+  const defaultCaipAccountAddresses = defaultAccounts.map(
+    ({ caipAccountId }) => caipAccountId,
   );
 
+  const [selectedCaipAccountAddresses, setSelectedCaipAccountAddresses] =
+    useState(defaultCaipAccountAddresses);
+
+  const selectedAccounts = allAccounts.filter(({ caipAccountId }) => {
+    return selectedCaipAccountAddresses.some((selectedCaipAccountId) => {
+      return isEqualCaseInsensitive(selectedCaipAccountId, caipAccountId);
```

Comment:
Not case sensitive

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/29887/commits/0739f4011195ed1854043ed013f56a589bb179e3